compiler/resolution/src/org/jetbrains/kotlin/resolve/calls/tower/ScopeTowerProcessors.kt

```diff
@@ -54,9 +54,31 @@ class SamePriorityCompositeScopeTowerProcessor<out C>(
 
 }
 
+interface SimpleScopeTowerProcessor<out C> : ScopeTowerProcessor<C> {
+    fun simpleProcess(data: TowerData): Collection<C>
+
+    override fun process(data: TowerData): List<Collection<C>> = listOfNotNull(simpleProcess(data).takeIf { it.isNotEmpty() })
+}
+
 internal abstract class AbstractSimpleScopeTowerProcessor<C : Candidate>(
     val candidateFactory: CandidateFactory<C>
-) : SimpleScopeTowerProcessor<C>
+) : SimpleScopeTowerProcessor<C> {
+    fun resolve(collector: Collection<CandidateWithBoundDispatchReceiver>, f:(Boolean)->Boolean, kind: ExplicitReceiverKind, receiver: ReceiverValueWithSmartCastInfo?) : Collection<C> {
```

Comment:
Regarding f, since receiver has a nullable type (i.e. ReceiverValueWithSmartCastInfo?), is it safe to assume receiver != null is equivalent to candidate.requiresExtensionReceiver true?

Reply:
I think it's safe.
Looks like, it should be `candidate.requiresExtensionReceiver == (receiver != null)`

Reply:
Gotcha, pushed as requested.